pennylane/spin/spin_hamiltonian.py

```diff
@@ -35,20 +35,21 @@ def transverse_ising(
 
         \hat{H} =  -J \sum_{<i,j>} \sigma_i^{z} \sigma_j^{z} - h\sum_{i} \sigma_{i}^{x}
 
-    where ``J`` is the coupling parameter defined for the Hamiltonian, ``h`` is the strength of the
-    transverse magnetic field and ``i,j`` represent the indices for neighbouring spins.
+    where :math:`J` is the coupling parameter defined for the Hamiltonian, :math:`h` is the strength of the
+    transverse magnetic field, :math:`<i,j>` represents the indices for neighbouring sites and
+    :math:`\sigma` is a Pauli operator.
 
     Args:
         lattice (str): Shape of the lattice. Input values can be ``'chain'``, ``'square'``,
-            ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, or ``'kagome'``.
+            ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, ``'kagome'``, or ``'diamond'``.
         n_cells (List[int]): Number of cells in each direction of the grid.
-        coupling (float or List[float] or List[math.array[float]]): Coupling between spins. It can
-            be a number, a list of length equal to ``neighbour_order`` or a square matrix of shape
+        coupling (float or tensor_like[float]): Coupling between spins. It should
+            be a number, an array of length equal to ``neighbour_order`` or a square matrix of shape
             ``(num_spins,  num_spins)``, where ``num_spins`` is the total number of spins. Default
```

Comment:
I just want to make sure here, are we keeping the matrix option?

Reply:
Yes.